MH4U Dex/CoreDataController.m

```diff
@@ -472,29 +484,70 @@ - (Quest *)questFromDictionary:(NSDictionary *)questDict withContext:(NSManagedO
     
     NSPredicate *monsterPredicate = [NSPredicate predicateWithFormat:@"%K == %@", MonsterAttributes.name, questDict[MHDQuestFirstMonsterKey]];
     // The same is the case with the following monster-fetching calls
-    quest.firstMonster = (Monster *)[self uniqueEntityWithEntityName:[Monster entityName]
-                                                       withPredicate:monsterPredicate
-                                                         withContext:context];
-    if (quest.firstMonster) {
-        //Continue on to the second, and so on.  By nesting this way, we can avoid unnecessary fetches
+    Monster *monster = (Monster *)[self uniqueEntityWithEntityName:[Monster entityName]
+                                                     withPredicate:monsterPredicate
+                                                       withContext:context];
+    if (monster) {
+        [quest.monsterSet addObject:monster];
         monsterPredicate = [NSPredicate predicateWithFormat:@"%K == %@", MonsterAttributes.name, questDict[MHDQuestSecondMonsterKey]];
-        quest.secondMonster = (Monster *)[self uniqueEntityWithEntityName:[Monster entityName]
-                                                            withPredicate:monsterPredicate
-                                                              withContext:context];
-        if (quest.secondMonster) {
-            monsterPredicate = [NSPredicate predicateWithFormat:@"%K == %@", MonsterAttributes.name, questDict[MHDQuestThirdMonsterKey]];
-            quest.thirdMonster = (Monster *)[self uniqueEntityWithEntityName:[Monster entityName]
+        Monster *secondMonster = (Monster *)[self uniqueEntityWithEntityName:[Monster entityName]
                                                                withPredicate:monsterPredicate
                                                                  withContext:context];
-            if (quest.thirdMonster) {
+        if (secondMonster) {
+            [quest.monsterSet addObject:secondMonster];
+            monsterPredicate = [NSPredicate predicateWithFormat:@"%K == %@", MonsterAttributes.name, questDict[MHDQuestThirdMonsterKey]];
```

Comment:
You addressed the `monster` variable, but you're still re-using the `monsterPredicate` throughout this method. I'd recommend changing that too.